db/hints/manager.cc

```diff
@@ -439,15 +439,15 @@ bool manager::check_dc_for(endpoint_id ep) const noexcept {
     }
 }
 
-void manager::drain_for(endpoint_id endpoint) {
+future<> manager::drain_for(endpoint_id endpoint) {
     if (!started() || stopping() || draining_all()) {
-        return;
+        return make_ready_future<>();
     }
 
     manager_logger.trace("on_leave_cluster: {} is removed/decommissioned", endpoint);
 
     // Future is waited on indirectly in `stop()` (via `_draining_eps_gate`).
```

Comment:
Comment is now outdated.

In general, dropping a future (even with a comment) is bad practice, so this is a good change.